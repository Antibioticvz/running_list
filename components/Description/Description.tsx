import React from 'react'
import { Colors, Text, TextArea, View } from 'react-native-ui-lib'
// import { Input } from 'react-native-elements'
      
interface Props {
    value: string,
    label: string,
    placeholder: string,
    onChange: (text: string) => void
}
const INPUT_SPACING = 10
const Description: React.FC<Props> = ({value, label, placeholder, onChange}) => {
    // const classes = useStyles()
    return (
      <>
        <Text text70 marginB-5>{label}</Text>
          <View
            style={{
              height: 150,
              borderWidth: 1,
              marginBottom: INPUT_SPACING,
              padding: 10,
              borderColor: Colors.dark60,
            }}
          >
            <TextArea {...{placeholder, value, onChangeText: onChange}}/>
          </View>
      </>
    )
}

export default Description
