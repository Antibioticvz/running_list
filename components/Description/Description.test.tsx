import React from 'react'
import 'react-native'
import { NativeModules } from 'react-native'
import renderer from 'react-test-renderer'
import Description from './Description'

NativeModules.StatusBarManager = {getHeight: jest.fn()}

describe('TextArea test', () => {
  it('Empty TextArea with title', () => {
    const description = renderer.create(
    <Description {...{
      label: 'Description',
      placeholder: 'What to expect',
      value: '',
      onChange: ()=>{},
    }}
    />).toJSON()
    expect(description).toMatchSnapshot()
  })

  it('TextArea with title and value', () => {
    const description = renderer.create(
    <Description {...{
      label: 'Description',
      placeholder: 'What to expect',
      value: 'My Value',
      onChange: ()=>{},
    }}
    />).toJSON()
    expect(description).toMatchSnapshot()
  })
})