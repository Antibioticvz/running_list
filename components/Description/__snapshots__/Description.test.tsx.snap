// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea test Empty TextArea with title 1`] = `
Array [
  <Text
    marginB-5={true}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "textAlign": "left",
        },
        Object {
          "fontFamily": undefined,
          "fontSize": 17,
          "fontWeight": "300",
          "lineHeight": 21,
        },
        undefined,
        Object {
          "marginBottom": 5,
        },
        undefined,
        undefined,
      ]
    }
    text70={true}
  >
    Description
  </Text>,
  <View
    style={
      Array [
        undefined,
        undefined,
        undefined,
        Object {},
        Object {},
        Object {},
        Object {
          "borderColor": "#C2C7CB",
          "borderWidth": 1,
          "height": 150,
          "marginBottom": 10,
          "padding": 10,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        multiline={true}
        onChangeText={[Function]}
        placeholder="What to expect"
        placeholderTextColor="#C2C7CB"
        rejectResponderTermination={true}
        style={
          Array [
            Object {
              "alignItems": "flex-start",
              "flex": 1,
              "justifyContent": "flex-start",
              "textAlignVertical": "top",
            },
            Object {
              "fontFamily": undefined,
              "fontSize": 17,
              "fontWeight": "300",
              "lineHeight": 21,
            },
          ]
        }
        underlineColorAndroid="transparent"
        validateOnBlur={true}
        value=""
      />
    </View>
  </View>,
]
`;

exports[`TextArea test TextArea with title and value 1`] = `
Array [
  <Text
    marginB-5={true}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "textAlign": "left",
        },
        Object {
          "fontFamily": undefined,
          "fontSize": 17,
          "fontWeight": "300",
          "lineHeight": 21,
        },
        undefined,
        Object {
          "marginBottom": 5,
        },
        undefined,
        undefined,
      ]
    }
    text70={true}
  >
    Description
  </Text>,
  <View
    style={
      Array [
        undefined,
        undefined,
        undefined,
        Object {},
        Object {},
        Object {},
        Object {
          "borderColor": "#C2C7CB",
          "borderWidth": 1,
          "height": 150,
          "marginBottom": 10,
          "padding": 10,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        multiline={true}
        onChangeText={[Function]}
        placeholder="What to expect"
        placeholderTextColor="#C2C7CB"
        rejectResponderTermination={true}
        style={
          Array [
            Object {
              "alignItems": "flex-start",
              "flex": 1,
              "justifyContent": "flex-start",
              "textAlignVertical": "top",
            },
            Object {
              "fontFamily": undefined,
              "fontSize": 17,
              "fontWeight": "300",
              "lineHeight": 21,
            },
          ]
        }
        underlineColorAndroid="transparent"
        validateOnBlur={true}
        value="My Value"
      />
    </View>
  </View>,
]
`;
